src/permissions.js

```diff
@@ -12,34 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-const makeChromeCall = (method, sites) => {
-  return new Promise((resolve, reject) => {
-    const origins = sites.map((site) => {
-      const a = document.createElement('a');
-      a.href = site['url'];
-      return a.origin + '/';
-    });
-    method({origins: origins}, (granted) => {
-      if (granted) {
-        resolve();
-      } else {
-        reject();
-      }
-    });
-  });
+import chromeAsync from './chromeasync';
+
+/**
+ * @param {{url: string}} site - URL.
+ * @return {string} The origin of the given site.
+ */
+const getOrigin = (site) => {
+  const a = document.createElement('a');
+  a.href = site.url;
+  return a.origin + '/';
 };
 
+/**
+ * @param {Promise} api - Promisified chrome API. See also chromeasync module.
```

Comment:
Good catch. Fixed. Along with the change, annotated type parameter to the Promise, too.